### PR TITLE
[lldb-dap][docs] Improve documentation for contributors

### PR DESCRIPTION
**Pull request by vogelsgesang**
This commit extends the developer docs for `lldb-dap`. It also adds a short "Contributing" section to the user-facing README.

Last but not least, it updates the `repository` in the package.json to point to the actual source of truth for the source code, instead of pointing to its mirrored repository. I hope that the VS Code Marketplace properly supports the `directory` property. Unfortunately, I have no way to test this before merging this Pull Request.

### CODE REVIEW

File: lldb/docs/resources/lldbdap.md [hunk 1]
```diff
@@ -1,16 +1,95 @@
...
-Protocol](https://microsoft.github.io/debug-adapter-protocol/). It can be
-installed as an extension for Visual Studio Code and other IDEs supporting DAP.
-The protocol is easy to run remotely and also can allow other tools and IDEs to
-get a full featured debugger with a well defined protocol.
-
-## Local Installation for Visual Studio Code
+# Contributing to LLDB-DAP
+
+This guide describes how to extend and contribute to lldb-dap.
+For documentation on how to use lldb-dap, see [lldb-dap's README](https://github.com/llvm/llvm-project/blob/release/19.x/lldb/tools/lldb-dap/README.md).
```
Comment (walter-erquinigo) [inline lldb/docs/resources/lldbdap.md:4]: better use `main` instead of `release/19.x`

**Review by walter-erquinigo — APPROVED** [review]
beautiful

File: lldb/docs/resources/lldbdap.md [hunk 2]
```diff
@@ -1,16 +1,95 @@
...
+
+### The Debug Adapter Protocol
+
+The most relevant resources for the Debug Adapter Protocol are:
+* [The overview](https://microsoft.github.io/debug-adapter-protocol/overview)
+  which provides a high-level introduction,
+* the [human-readable specification](https://microsoft.github.io/debug-adapter-protocol/specification), and
+* the [JSON-schema specification](https://github.com/microsoft/debug-adapter-protocol/blob/main/debugAdapterProtocol.json).
+
+Sometimes, lldb-dap also adds proprietary extensions to the protocol. To take
```
Comment (ashgti) [inline lldb/docs/resources/lldbdap.md:64]: nit: maybe something along the lines of

lldb-dap includes some additional non-standard extensions to the protocol.

File: lldb/docs/resources/lldbdap.md [hunk 3]
```diff
@@ -1,16 +1,95 @@
...
+advantage of those proprietary protocol extensions, IDE-specific support code
```
Comment (ashgti) [inline lldb/docs/resources/lldbdap.md:65]: I don't think `proprietary` is the right word here, how about custom extenstions? or non-standard extensions?

File: lldb/docs/resources/lldbdap.md [hunk 4]
```diff
@@ -1,16 +1,95 @@
...
+is needed, usually inside the VS Code extension. When adding a new extension,
+please first look through the [issue tracker of the Debug Adapter
+Protocol](https://github.com/microsoft/debug-adapter-protocol/issues) to check
+if there already is a proposal serving your use case. If so, try to take
+inspiration from it. If not, consider opening an upstream issue.
+
+To avoid naming collisions with potential future extensions of the Debug
+Adapter protocol, all proprietary extensions should use the prefix
```
Comment (ashgti) [inline lldb/docs/resources/lldbdap.md:73]: ditto

**Review by ashgti — APPROVED** [review]

**Pull request merged**